shared/events.js

```diff
@@ -377,11 +389,10 @@ export class DeleteAttribute extends HashableAction {
 
 export class MailboxContract extends HashableContract {
   static vuex = MailboxContract.Vuex({
-    state: { messages: [], nextId: 0 },
+    state: { messages: [] },
     mutations: {
-      // TODO: verify the correctness of 'nextId' in more detail later
-      PostMessage (state, data) { state.messages.push({...data, id: state.nextId++}) },
-      AuthorizeSender (state, data) { state.authorizations[AuthorizeSender.authorization.name].data = data.sender }
+      PostMessage (state, {data, hash}) { state.messages.push({...data, hash}) },
```

Comment:
Why would it? Shouldn't we know that in advance an not design the data that way. You seemed to have a problem with modifying the data object to include the hash before. So what do you want?